kafka-client/src/main/java/dev/responsive/kafka/internal/stores/ResponsiveWindowStore.java

```diff
@@ -81,14 +94,26 @@ public void init(final StateStoreContext storeContext, final StateStore root) {
     try {
       log.info("Initializing state store");
 
+      final var appConfigs = storeContext.appConfigs();
+      final ResponsiveConfig config = ResponsiveConfig.responsiveConfig(appConfigs);
+
       context = storeContext;
       windowOperations = SegmentedOperations.create(
           name,
           storeContext,
           params,
+          appConfigs,
+          config,
           window -> window.windowStartMs >= minValidTimestamp()
       );
 
+      enableBloomFilter = config.getBoolean(WINDOW_BLOOM_FILTER_ENABLED_CONFIG);
+      if (enableBloomFilter) {
+        expectedKeysPerWindow = config.getLong(WINDOW_BLOOM_FILTER_EXPECTED_KEYS_CONFIG);
+        fpp = config.getDouble(WINDOW_BLOOM_FILTER_FPP_CONFIG);
+        bloomFilter = BloomFilter.create(Funnels.byteArrayFunnel(), expectedKeysPerWindow, fpp);
```

Comment:
I think the way this is implemented now we're susceptible to getting false-negative reads from the filter in a situation like:
Otherwise in a scenario like the following, we'll get false negatives from the filter:
1. start a window
2. write some records
3. crash and recover
4. read from filter

We need to make sure that every record that we write into the current window is also added to the bloom filter. The simplest way I can think to do this we would be to only create and use a bloom filter once we've observed the window start time change. We'll get poorer performance for the first window after a restart, but I think that's ok for now since we're optimizing for a use case with a relatively short window.

Reply:
Yes, this is definitely the case. One of the tests covers it and is failing, I just haven't pushed a fix yet. My plan is to just ignore the filter unless we are certain it contains the entire window so far. 

I think that's probably fine for cases with smaller window sizes, we can probably go 15 minutes without a restart/rebalance for example. But we'll definitely need something a bit more sophisticated for longer window sizes (or account for that in the assignor to avoid moving tasks with windowed state, if possible)